### ISSUE TITLE
Fatal exception in F# unused code highlighter

### ISSUE THREAD

**Issue by nosami** [mono/monodevelop#4351]
```
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentNullException: Value cannot be null.
Parameter name: segment
  at MonoDevelop.Ide.Editor.TextEditor.GetTextAt (MonoDevelop.Core.Text.ISegment segment) [0x0000e] in /Users/builder/data/lanes/432/0d559012/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs:816
  at MonoDevelop.Ide.Editor.TextEditor.GetLineText (System.Int32 line, System.Boolean includeDelimiter) [0x00008] in /Users/builder/data/lanes/432/0d559012/source/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs:597
  at MonoDevelop.FSharp.highlightUnusedCode+getUnusedCode@93-1.Invoke (System.Int32 lineNum) [0x00000] in /Users/builder/data/lanes/432/0d559012/source/monodevelop/main/external/fsharpbinding/MonoDevelop.FSharpBinding/HighlightUnusedCode.fs:93
```

> VS bug [#591145](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/591145)

**Comment by nosami** [comment]
https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor/TextEditor.cs?utf8=%E2%9C%93#L816

Not sure how ReadOnlyTextDocument could be null

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-17] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #4416 merged by: Jason Imison  
 Author: mono  
 Commit: 5063d19709d8b719a3019892c7f923acda870cfb (xamarin/monodevelop)
